Shared/mods/deathmatch/logic/CLodModels.cpp

```diff
@@ -0,0 +1,493 @@
+/*****************************************************************************
+ *
+ *  PROJECT:     Multi Theft Auto v1.0
+ *               (Shared logic for modifications)
+ *  LICENSE:     See LICENSE in the top level directory
+ *  FILE:        mods/shared_logic/CLodModels.cpp
+ *
+ *****************************************************************************/
+
+#include "StdInc.h"
+#include "CLodModels.h"
+
+// This associates Object Model ID to its Object-LOD Model ID, according to the game's IPL files (including the binary ones).
+
+std::unique_ptr<const std::map<std::uint32_t, std::uint32_t>> CLodModels::LOD_MODELS =
+    std::make_unique<const std::map<std::uint32_t, std::uint32_t>>(std::map<std::uint32_t, std::uint32_t>{
```

Comment:
why unique_ptr?
Just store a constexpr map to it:
```cpp
const std::unordered_map<std::uint32_t, std::uint32_t> LOD_MODELS = {
// ...
};

```

Reply:
constexpr complained there was too much data (the array is huge), this is perfect in terms of memory management

Reply:
it will be allocated either way

Reply:
ok let me try

Reply:
after internal discussions unique_ptr is necessary here due to the amount of data. Stack will be filled so a heap need to be used here